recipes/octomap/all/conanfile.py

```diff
@@ -0,0 +1,67 @@
+import os
+
+from conans import ConanFile, CMake, tools
+
+class OctomapConan(ConanFile):
+    name = "octomap"
+    description = "An Efficient Probabilistic 3D Mapping Framework Based on Octrees."
+    license = "BSD-3-Clause"
+    topics = ("conan", "octomap", "octree", "3d", "robotics")
+    homepage = "https://github.com/OctoMap/octomap"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "openmp": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "openmp": False
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "octomap","CMakeLists.txt"),
+                              "SET( BASE_DIR ${CMAKE_SOURCE_DIR} )",
+                              "SET( BASE_DIR ${CMAKE_BINARY_DIR} )")
+        cmake = CMake(self)
+        cmake.definitions["OCTOMAP_OMP"] = self.options.openmp
+        cmake.configure(build_folder=self._build_subfolder)
+        cmake.build(target="octomap" if self.options.shared else "octomap-static")
```

Comment:
why don't you chdir to octomap folder instead and build from there? So you can use cmake.install too.

Reply:
Isn't it already the case? CMake wrapper add "source_folder/octomap" subdirectory. Anyway, calling install will build both static and shared libs:
https://github.com/OctoMap/octomap/blob/v1.9.3/octomap/src/CMakeLists.txt

Reply:
hmm, I didn't see that, sorry.